### ISSUE TITLE
Xonsh 0.8.11 fails on startup

### ISSUE THREAD

**Issue by kbd** [xonsh/xonsh#3034]
```
⚡~$ pip3 install --force xonsh
Collecting xonsh
Installing collected packages: xonsh
  Found existing installation: xonsh 0.8.11
    Uninstalling xonsh-0.8.11:
      Successfully uninstalled xonsh-0.8.11
Successfully installed xonsh-0.8.11
⚡~$ xonsh --version
xonsh/0.8.11
⚡~$ xonsh
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 24004, in main
    args = premain(argv)
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 23955, in premain
    env = start_services(shell_kwargs, args)
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 23890, in start_services
    cacheall=shell_kwargs.get("cacheall", False),
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 23021, in __init__
    load_builtins(execer=self, ctx=xonsh_ctx)
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 22592, in load_builtins
    builtins.__xonsh__.load(execer=execer, ctx=ctx)
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 22668, in load
    self.env = Env(default_env())
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 19110, in __init__
    self[key] = val
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 19272, in __setitem__
    val = ensurer.convert(val)
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 18218, in convert
    return cls.fromstring(x)
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 18181, in fromstring
    esc, "default", reversed_style=reversed_default
  File "/usr/local/lib/python3.7/site-packages/xonsh/__amalgam__.py", line 6986, in ansi_color_escape_code_to_name
    ec = ANSI_ESCAPE_CODE_RE.match(escape_code).group(2)
AttributeError: 'NoneType' object has no attribute 'group'
Xonsh encountered an issue during launch
Failback to /bin/bash
bash: shopt: globstar: invalid shell option name
bash: shopt: autocd: invalid shell option name
[
a few more failures because my bashrc depends on current
bash at /usr/local/bin/bash, not the old system bash
]
bash-3.2$ exit
```

I previously had 0.8.6 IIRC (which worked) and just installed 0.8.11. Was gonna use some low-energy time to configure my Xonsh config to get parity with Zsh to see if I could switch to Xonsh and then this happened :) Running on Mac Mojave. Lemme know if I can do anything else to help debug.

**Comment by scopatz** [comment]
Thanks for reporting  @kbd! What is the output of the `dircolors` command?

**Comment by kbd** [comment]
```
⚡~$ gdircolors
LS_COLORS='rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:';
export LS_COLORS
```

Most of which was taken from https://github.com/trapd00r/LS_COLORS forever ago and I customized a bit.

**Comment by scopatz** [comment]
There is a potential fix in #3045, if you'd like to check that out

**Comment by scopatz** [comment]
Thanks for the LS_COLORS link too, that helped a lot.
